source/s3_meta_request.c

```diff
@@ -208,9 +177,69 @@ int aws_s3_meta_request_init_base(
 }
 
 void aws_s3_meta_request_cancel(struct aws_s3_meta_request *meta_request) {
+    aws_s3_meta_request_lock_synced_data(meta_request);
+    aws_s3_meta_request_set_fail_synced(meta_request, NULL, AWS_ERROR_S3_CANCELED);
+    aws_s3_meta_request_unlock_synced_data(meta_request);
+}
+
+void aws_s3_meta_request_set_fail_synced(
```

Comment:
confused why these are named `_synced` when there's no non-synced version and they don't hold any locks

Reply:
oh, does `_synced` indicate it's only to be called inside a critical section?

Reply:
Yeah, I was trying to indicate what functions assume a lock is held, and then they're meant to each have an ASSERT_SYNCED_DATA_LOCK_HELD that actually check if the mutex is locked.